setoff_workflows/setoff_workflows.py

```diff
@@ -0,0 +1,2622 @@
+            )
+            dx_cmds.append(SWConfig.UPLOAD_ARGS["depends_list"])
+
+            if self.samples_obj.pipeline == "pipe":
+                # Add to gatk depends list because RPKM must depend only upon the
+                # sample workflows completing successfully, whilst other downstream
+                # apps depend on all prior jobs completing succesfully
+                dx_cmds.append(SWConfig.UPLOAD_ARGS["depends_list_gatk"])
+
+            if self.samples_obj.pipeline in ["wes", "pipe"]:
+                decision_support_upload_cmds.append(
+                    self.samples_obj.samples_dict[sample_name][
+                        "decision_support_upload_cmd"
+                    ]
+                )
+        return dx_cmds, decision_support_upload_cmds
+
+    def return_wes_runwide_cmds(self) -> list:
+        """
+        Collect runwide commands for WES runs as a list. This includes peddy.
+            :return cmd_list (list):    List of runwide commands for WES runs
+        """
+        return [self.create_peddy_cmd(), SWConfig.UPLOAD_ARGS["depends_list"]]
+
+    def create_peddy_cmd(self) -> str:
+        """
+        Build dx run command to run peddy for the project. Run once at the
+        end of a WES run and downloads required files from the project
+            :return (str):  Dx run command for peddy app
+        """
+        self.rf_obj.rf_loggers.sw.info(
+            self.rf_obj.rf_loggers.sw.log_msgs["building_cmd"],
+            "peddy",
+            self.rf_obj.runfolder_name,
+        )
+        return " ".join(
+            [
+                f'{SWConfig.DX_CMDS["peddy"]}Peddy',
+                SWConfig.APP_INPUTS["peddy"]["project_name"],
+                SWConfig.UPLOAD_ARGS["depends"],
+                SWConfig.UPLOAD_ARGS["dest"],
+                SWConfig.UPLOAD_ARGS["token"] % self.dnanexus_auth,
+            ]
+        )
+
+    def return_pipe_runwide_cmds(self) -> list:
+        """
+        Collect runwide commands for PIPE runs as a list. This includes RPKM
+        (single command per vcp panel)
+            :return cmd_list (list):    List of runwide commands for PIPE runs
+        """
+        cmd_list = []
+        for core_panel in ["vcp1", "vcp2", "vcp3"]:
+            if core_panel in (
+                [
+                    self.samples_obj.samples_dict[k]["panel_settings"]["panel_name"]
+                    for k, v in self.samples_obj.samples_dict.items()
+                ]
+            ):
+                core_panel_pannos = [
+                    self.samples_obj.samples_dict[k]["pannum"]
+                    for k, v in self.samples_obj.samples_dict.items()
+                    if self.samples_obj.samples_dict[k]["panel_settings"]["panel_name"]
+                    == core_panel
+                ]
+                # Make sure there are enough samples for exome depth and RPKM
+                if len(core_panel_pannos) >= 3:
```

Comment:
If there are <3, can something go to the log that says ED/RPKM not run because not enough samples?

Reply:
Done - good spot